src/black/linegen.py

```diff
@@ -1074,7 +1075,14 @@ def bracket_split_build_line(
             no_commas = (
                 original.is_def
                 and opening_bracket.value == "("
-                and not any(leaf.type == token.COMMA for leaf in leaves)
+                and not any(
+                    leaf.type == token.COMMA
+                    and (
+                        Preview.typed_params_trailing_comma not in original.mode
+                        or not is_part_of_annotation(leaf)
```

Comment:
Can we use `is_part_of_annotation` to simplify some of the logic in the next check too? Lines 1092-1099?

Reply:
I tried, but it ended up adding a lot of unrelated changes to this PR since `is_part_of_annotation` currently checks for both return and parameter annotations. I can do that refactor in a future PR.